lib/tooling/llvm-mca-tool.ts

```diff
@@ -47,21 +47,59 @@ export class LLVMMcaTool extends BaseTool {
 
     override async runTool(compilationInfo: Record<any, any>, inputFilepath?: string, args?: string[]) {
         const isets = new InstructionSets();
-        const target = isets.getInstructionSetTarget(compilationInfo.compiler.instructionSet);
-        if (target != null) args?.push('-mtriple=' + target);
+        let target = isets.getInstructionSetTarget(compilationInfo.compiler.instructionSet);
+        const prependArgs: string[] = [];
 
+        // Check if compiler target is overridden with --target=<foo> or -target <foo>.
+        let argIsTarget = false;
         for (const arg of compilationInfo.options) {
-            if (arg.startsWith('-mcpu') || arg.startsWith('-march') || arg.startsWith('-mtriple')) {
-                args?.push(arg);
+            if (arg.startsWith('--target=')) target = arg.replace(/^--target=/, '');
+            if (argIsTarget) {
+                target = arg;
+                argIsTarget = false;
+            }
+            if (arg === '-target') argIsTarget = true;
+        }
+
+        if (target != null) prependArgs.push('-mtriple=' + target);
```

Comment:
should this be `!==` here? or are you trying to do something cunning in typescript/javascript that I don't understand? Is this materially different to `if (target) ...` ?

Reply:
The choice of expression was arbitrary. I've written it as `if (target)` as I understand you are suggesting.